contracts/contracts/core/PreconfManagerStorage.sol

```diff
@@ -36,6 +33,10 @@ abstract contract PreconfManagerStorage {
     /// @dev Only stores valid unopened commitments
     mapping(bytes32 => IPreconfManager.UnopenedCommitment) public unopenedCommitments;
 
+    /// @dev Mapping from txn hash plus bidder address to if it has been processed
+    /// @dev Used to prevent rewarding/slashing multiple times
+    mapping(bytes32 => bool) public processedTxnHashes;
+
     /// @dev See https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#storage-gaps
     uint256[48] private __gap;
```

Comment:
reduce to 47 when we add more datapoints

Reply:
Mapping is not being taken into account here.

Reply:
N/a to PR in main anyways